### PR TITLE
Cleanup consensus metrics

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

Currently our metrics logic in the snowman consensus engine is repetitive and makes the code more difficult to understand. Additionally, it can result in us updating the same metric multiple times in the same call... Even though we really only need to update them once at the end.

## How this works

Updates metrics once at the end of the engine modifications.

## How this was tested

N/A

### CODE REVIEW

File: snow/engine/snowman/transitive.go [hunk 1]
```diff
@@ -553,7 +549,7 @@ func (t *Transitive) Start(ctx context.Context, startReqID uint32) error {
 		return fmt.Errorf("failed to notify VM that consensus is starting: %w",
 			err)
 	}
-	return nil
+	return t.executeDeferredWork(ctx)
```
Comment (StephenButtolph) [inline snow/engine/snowman/transitive.go:552]: Added here because of the call to `t.deliver` above.

File: snow/engine/snowman/transitive.go [hunk 2]
```diff
@@ -584,7 +580,19 @@ func (t *Transitive) HealthCheck(ctx context.Context) (interface{}, error) {
 	return intf, fmt.Errorf("vm: %w ; consensus: %w", vmErr, consensusErr)
 }
 
-func (t *Transitive) GetBlock(ctx context.Context, blkID ids.ID) (snowman.Block, error) {
+func (t *Transitive) executeDeferredWork(ctx context.Context) error {
```
Comment (StephenButtolph) [inline snow/engine/snowman/transitive.go:583]: This function is going to get larger as I continue to cleanup the logic here.

File: snow/engine/snowman/transitive.go [hunk 3]
```diff
@@ -584,7 +580,19 @@ func (t *Transitive) HealthCheck(ctx context.Context) (interface{}, error) {
...
+	}
+
+	t.metrics.numRequests.Set(float64(t.blkReqs.Len()))
+	t.metrics.numBlocked.Set(float64(len(t.pending)))
+	t.metrics.numBlockers.Set(float64(t.blocked.Len()))
+	t.metrics.numNonVerifieds.Set(float64(t.nonVerifieds.Len()))
+	return nil
+}
+
+func (t *Transitive) getBlock(ctx context.Context, blkID ids.ID) (snowman.Block, error) {
```
Comment (StephenButtolph) [inline snow/engine/snowman/transitive.go:595]: Minor unrelated cleanup - but this method didn't need to be exported anymore.

File: snow/engine/snowman/transitive.go [hunk 4]
```diff
@@ -584,7 +580,19 @@ func (t *Transitive) HealthCheck(ctx context.Context) (interface{}, error) {
...
-func (t *Transitive) GetBlock(ctx context.Context, blkID ids.ID) (snowman.Block, error) {
+func (t *Transitive) executeDeferredWork(ctx context.Context) error {
+	if err := t.buildBlocks(ctx); err != nil {
+		return err
+	}
+
+	t.metrics.numRequests.Set(float64(t.blkReqs.Len()))
+	t.metrics.numBlocked.Set(float64(len(t.pending)))
+	t.metrics.numBlockers.Set(float64(t.blocked.Len()))
+	t.metrics.numNonVerifieds.Set(float64(t.nonVerifieds.Len()))
```
Comment (StephenButtolph) [inline snow/engine/snowman/transitive.go:591]: This is now the only location these gauges are set.

**Review by joshua-kim — APPROVED** [review]

**Review by abi87 — APPROVED** [review]

**Pull request merged**